lib/cloudant.js

```diff
@@ -522,6 +524,19 @@ Cloudant.prototype.updateAttributes = function(model, id, data, options, cb) {
 };
 
 /**
+ * If input is a model instance, convert to a plain JSON object
+ * The input would be a model instance when the request is made from
+ * REST endpoint as remoting converts it to model if endpoint expects a
+ * model instance
+ */
+
+function getPlainJSONData(model, data) {
+  if (this._models[model] && data instanceof this._models[model].model) {
+    return data.toJSON();
+  }
+  return data;
+}
```

Comment:
AFAIK, this function i.e. parsing data from model instance is not present in the base connector and in other connectors, there is some sort of cleanup built-in for input data.
Thus, defined parsing of input data here in the connector.